src/install.rs

```diff
@@ -136,3 +150,95 @@ fn chmod_x(file: &PathBuf) {
         error!("Failed to run chmod +x: {}", e)
     };
 }
+
+fn mac_process_zip(archive_path: String) {
+    info!("Mac zip postprocessing starting...");
+
+    // We have to do the unzipping via shell because the tool omits the app directory and doesn't set the execution flag properly
```

Comment:
I'd prefer to add a flag to the respective unzip function so it doesn't always ignore the toplevel directory (remind to do that if you don't want to - it's pretty simple). The chmod stuff can be done the same way linux does it (L147).

Reply:
Please do that, I am really struggeling with Rust and it takes me way too long to achieve anything with it. 
One thing noteworthy - when using the unzip command of MacOS, the execution bits are properly set, hence no subsequent command in my code. I just forgot to update the comment. If you handle it the way you handle the linux and windows zips, the execution bits have to be set afterwards.